torchgeo/datasets/l8biome.py

```diff
@@ -0,0 +1,265 @@
+                don't match
+        """
+        self.root = root
+        self.bands = bands
+        self.band_indexes = [self.bands.index(i) for i in self.bands]
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        super().__init__(root, crs=crs, res=res, bands=bands, transforms=transforms, cache=cache)
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted files already exist
+        pathname = os.path.join(self.root, "**", self.filename_glob)
+        for fname in glob.iglob(pathname, recursive=True):
+            if not fname.endswith(".tar.gz"):
+                return
+
+        # Check if the tar.gz files have already been downloaded
+        pathname = os.path.join(self.root, "*.tar.gz")
+        if glob.glob(pathname):
+            self._extract()
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        for biome, md5 in self.filenames_to_md5.items():
+            download_url(
+                self.url.format(biome), self.root, md5=md5 if self.checksum else None
+            )
+
+    def _extract(self) -> None:
+        """Extract the dataset."""
+        pathname = os.path.join(self.root, "*.tar.gz")
+        for tarfile in glob.iglob(pathname):
+            extract_archive(tarfile)
+
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+        Returns:
+            sample of image, mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        img_filepaths = cast(List[str], [hit.object for hit in hits])
+
+        # print(img_filepaths)
+        mask_filepaths = []
+
+        for path in img_filepaths:
+            mask_file_path = path.replace("B2.TIF", "fixedmask.img")
+            mask_filepaths.append(mask_file_path)
+
+        if not img_filepaths:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+        
+        img = self._merge_files(img_filepaths, query, self.band_indexes)
```

Comment:
Remove `self.band_indexes` and that should solve your latest bug. `band_indexes` is only needed for `separate_files = False`.